File: ObjC/HSSDKXObjcSample/HSSDKXObjcSample/HelpshiftEventsHandler.m
//
//  HelpshiftEventsHandler.m
//  HSSDKXObjcSample
//
//  Created by Sagar Dagdu on 24/05/21.
//  Copyright © 2021 Sagar Dagdu. All rights reserved.
//

#import "HelpshiftEventsHandler.h"

@implementation HelpshiftEventsHandler

+ (instancetype)sharedInstance {
    static HelpshiftEventsHandler *sharedInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[HelpshiftEventsHandler alloc] init];
    });
    
    return sharedInstance;
}

#pragma mark Helpshift Delegate Implementation

- (void)handleHelpshiftEvent:(NSString *)eventName withData:(NSDictionary *)data {
    //TODO: Handle events from HelpshiftX SDK here.
    
    NSLog(@"Helpshift Event: %@, event data: %@", eventName, data);
}

- (void)authenticationFailedForUserWithReason:(HelpshiftAuthenticationFailureReason)reason {
    //TODO: Handle authentication failed.
    
    NSLog(@"Helpshift: Failed to authenticate user, reason: %lu", (unsigned long)reason);
}

@end

File: Swift/HSSDKXSwiftSample/HSSDKXSwiftSample/HelpshiftEventsHandler.swift
//
//  HelpshiftEventsHandler.swift
//  HSSDKXSwiftSample
//
//  Created by Sagar Dagdu on 28/05/21.
//  Copyright © 2021 Sagar Dagdu. All rights reserved.
//

import UIKit
import HelpshiftX

final class HelpshiftEventsHandler: NSObject, HelpshiftDelegate {
    
    static let shared = HelpshiftEventsHandler()
    
    private override init() {
        
    }
    
    func handleHelpshiftEvent(_ eventName: String, withData data: [AnyHashable : Any]?) {
        //TODO: Handle events from HelpshiftX SDK here.
        let eventData = data ?? [AnyHashable:Any]()
        print("Helpshift Event: \(eventName), event data: \(eventData)")
    }
    
    func authenticationFailedForUser(with reason: HelpshiftAuthenticationFailureReason) {
        //TODO: Handle authentication failed.
        print("Helpshift: Failed to authenticate user, reason: \(reason)")
    }
    
    
}
